### ISSUE TITLE
Shape mismatch after NaN filter: posture.primary_wavelength

### ISSUE THREAD

**Issue by MichaelCurrie** [openworm/movement_validation#114]
Running `validate_features.py`, I get the following output:

```
Calculating Morphology Features
Calculating Locomotion Features
Calculating Posture Features
Calculating Path Features

Comparison of computed features to those computed with old Matlab code
Locomotion: True
Shape mismatch after NaN filter: posture.primary_wavelength
Shape mismatch after NaN filter: posture.secondary_wavelength
Posture: False
Morphology: True
Path: True
```

I get this output whether or not `user_config.py` has
`PERFORM_SLOW_ECCENTRICITY_CALC = False` or
`PERFORM_SLOW_ECCENTRICITY_CALC = True`.

**Comment by MichaelCurrie** [comment]
Now I'm getting a different error.  Jim I hope you don't mind me assigning you to this, feel free to unassign if you prefer not to do it.

```
Calculating Morphology Features
Calculating Locomotion Features
Calculating Posture Features
Calculating Path Features

Comparison of computed features to those computed with old Matlab code
Shape mismatch (1,) vs (2,): locomotion.turns.upsilons.start_frames
Mismatch in locomotion.turns events
Locomotion: False
Posture: None
Morphology: True
Path: True

Done validating features
Time elapsed: 47.69 seconds
```

**Comment by JimHokanson** [comment]
This turned out to be an error introduced by @MichaelCurrie :stuck_out_tongue:

In all fairness one of us had written != None when in reality is not None is the appropriate way of writing this. It had been changed to look for numpy arrays which doesn't work with scalar inputs

**Comment by JimHokanson** [comment]
We need to change the example to throw an example if the comparisons fail for Travis testing, see #146

**Comment by MichaelCurrie** [comment]
Oops!  Nice catch.

**Comment by MichaelCurrie** [comment]
A remaining error: Posture returns None (see above)

The cause is that `posture.bends.__eq__` was not implemented.

I fixed this with 3dfde14a1325251ca1ab7f6e0c3aa387fa999d16, now the tests pass, see #146.
